### PR TITLE
Improve tests for BlockWorkerDataReader

### PR DESCRIPTION
**Pull request by apc999**
Followup of https://github.com/Alluxio/alluxio/pull/13282/ to improve tests of `BlockWorkerDataReader`

### CODE REVIEW

**Review by LuQQiu — APPROVED** [review]
LGTM! Appreciate for giving out this great example! I got your idea of how to implement the test and how to test the real behaviors. If you haven't started working on the DefaultBlockWorkerTest, I plan to modify that unit test. If you started, I will choose another test to practice the unit test writing.

File: core/server/worker/src/test/java/alluxio/worker/block/stream/BlockWorkerDataReaderTest.java [hunk 1]
```diff
@@ -11,97 +11,149 @@
...
-    AlluxioConfiguration conf
-        = new InstancedConfiguration(ConfigurationUtils.defaults());
-    InStreamOptions options = new InStreamOptions(dummyStatus,
-        FileSystemOptions.openFileDefaults(conf), conf);
-    mDataReaderFactory = new BlockWorkerDataReader
-        .Factory(mBlockWorker, BLOCK_ID, CHUNK_SIZE, options);
+    mBlockMasterClient = mock(BlockMasterClient.class);
+    mBlockMasterClientPool = spy(new BlockMasterClientPool());
+    when(mBlockMasterClientPool.createNewResource()).thenReturn(mBlockMasterClient);
+    mBlockStore = spy(new TieredBlockStore());
```
[the remark this answers is not in the document]
Comment (apc999) [reply core/server/worker/src/test/java/alluxio/worker/block/stream/BlockWorkerDataReaderTest.java]: fixed. thanks!

**Comment by LuQQiu** [comment]
Tests failures look irrelavant, rerunning 
Errors: 
Error: 2.328 [ERROR]   ChgrpCommandIntegrationTest>AbstractFileSystemShellTest.beforeClass:123 » Runtime

Error: 2.813 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project alluxio-core-transport: Compilation failure
Error: 2.815 [ERROR] /usr/src/alluxio/core/transport/target/generated-sources/protobuf/grpc-java/alluxio/grpc/BlockMasterWorkerServiceGrpc.java:[527,54] incompatible types: inference variable RequestT has incompatible bounds
Error: 2.816 [ERROR]     equality constraints: ResponseT,alluxio.grpc.CommitBlockInUfsPResponse,RespT,ReqT,alluxio.grpc.CommitBlockInUfsPRequest
Error: 2.816 [ERROR]     lower bounds: alluxio.grpc.CommitBlockInUfsPRequest

Error: 3.364 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project alluxio-table-base: Unable to parse configuration of mojo org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile for parameter compilePath: Cannot evaluate expression '${project.compileClasspathElements}' for configuration entry 'compilePath': Error evaluating plugin parameter expression: project.compileClasspathElements: InvocationTargetException: NullPointerException -> [Help 1]

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**